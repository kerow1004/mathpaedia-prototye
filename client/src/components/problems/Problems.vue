<template>
  <b-list-group id="problems-list">
    <problem v-for="problem in problems" :problem="problem" :key="problem._id"></problem>
  </b-list-group>
</template>

<script>
import Problem from './Problem';

export default {
  name: 'Problems',
  components: {
    problem: Problem,
  },
  computed: {
    problems() {
      return this.$store.getters.problems;
    },
  },
  created() {
    this.$log.info('[Problems] created');
    this.$store.dispatch('getProblems');

    /* eslint-disable */
    this.$nextTick(() => {
      MathJax.Hub.Queue(['Typeset', MathJax.Hub, 'problems-list']);
    });
    /* eslint-enable */
  },
};
</script>

<style scoped>

</style>
