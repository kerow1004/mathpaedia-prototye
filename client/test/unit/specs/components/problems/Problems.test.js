import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueLogger from 'vuejs-logger';
import BootstrapVue from 'bootstrap-vue';
import Problems from '@/components/problems/Problems';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueLogger);
localVue.use(BootstrapVue);

describe('Math Problems.vue', () => {
  const problems = [
    {
      sentence: 'math problem sentence 1',
      picture: 'http://picture.math.com/1234561',
      choices: ['11', '12', '13', '14', '15'],
    },
    {
      sentence: 'math problem sentence 2',
      picture: 'http://picture.math.com/1234562',
      choices: ['21', '22', '23', '24', '25'],
    },
  ];

  const oriMathJax = global.MathJax;

  let actions;
  let getters;
  let store;
  let wrapper;

  beforeAll(() => {
    global.MathJax = {
      Hub: {
        Queue: jest.fn(),
      },
    };

    actions = {
      getProblems: jest.fn(),
    };

    getters = {
      problems: jest.fn(() => problems),
    };

    store = new Vuex.Store({
      state: {},
      actions,
      getters,
    });

    wrapper = mount(Problems, {
      localVue,
      store,
    });
  });

  afterAll(() => {
    global.MathJax = oriMathJax;
  });

  test('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('calls store action "getProblems" when created', () => {
    expect(actions.getProblems).toHaveBeenCalled();
  });

  test('calls store getters "problems" when changed', () => {
    expect(getters.problems).toHaveBeenCalled();
  });
});
