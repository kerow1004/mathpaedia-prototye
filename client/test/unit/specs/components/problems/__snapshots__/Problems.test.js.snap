// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Math Problems.vue has the expected html structure 1`] = `
<div
  class="list-group"
  id="problems-list"
>
  <div
    class="list-group-item"
  >
    <!---->
  </div>
  <div
    class="list-group-item"
  >
    <!---->
  </div>
</div>
`;
